# origin: Debian (herbert)
# cset: n/a
# inclusion: not submitted
# description: don't perform FP operations in amd8111e time calculations
# revision date: 2004-09-04

diff -urN kernel-source-2.4.26/drivers/net/amd8111e.h kernel-source-2.4.26-1/drivers/net/amd8111e.h
--- kernel-source-2.4.26/drivers/net/amd8111e.h	2003-08-25 21:44:42.000000000 +1000
+++ kernel-source-2.4.26-1/drivers/net/amd8111e.h	2003-10-19 20:40:36.000000000 +1000
@@ -606,7 +606,7 @@
 /* ipg parameters */
 #define DEFAULT_IPG			0x60
 #define IFS1_DELTA			36
-#define	IPG_CONVERGE_TIME 0.5
+#define	IPG_CONVERGE_JIFFIES		(HZ / 2)
 #define	IPG_STABLE_TIME	5
 #define	MIN_IPG	96
 #define	MAX_IPG	255

--- kernel-source-2.4.27-2.4.27/drivers/net/amd8111e.c~	2004-08-11 10:35:57.000000000 -0700
+++ kernel-source-2.4.27-2.4.27/drivers/net/amd8111e.c	2004-08-11 10:37:10.000000000 -0700
@@ -1726,7 +1726,7 @@
 	/* Restart ipg timer */
 	if(lp->options & OPTION_DYN_IPG_ENABLE)	        
 		mod_timer(&lp->ipg_data.ipg_timer, 
-				jiffies + (IPG_CONVERGE_TIME * HZ));
+				jiffies + IPG_CONVERGE_JIFFIES);
 	spin_unlock_irq(&lp->lock);
 
 	return 0;
@@ -1800,7 +1800,7 @@
 		writew((u32)tmp_ipg, mmio + IPG); 
 		writew((u32)(tmp_ipg - IFS1_DELTA), mmio + IFS1); 
 	}
-	 mod_timer(&lp->ipg_data.ipg_timer, jiffies + (IPG_CONVERGE_TIME * HZ));
+	 mod_timer(&lp->ipg_data.ipg_timer, jiffies + IPG_CONVERGE_JIFFIES);
 	return;
 
 }
@@ -1938,7 +1938,7 @@
 		lp->ipg_data.ipg_timer.data = (unsigned long) dev;
 		lp->ipg_data.ipg_timer.function = (void *)&amd8111e_config_ipg;
 		lp->ipg_data.ipg_timer.expires = jiffies + 
-						 IPG_CONVERGE_TIME * HZ;
+						 IPG_CONVERGE_JIFFIES;
 		lp->ipg_data.ipg = DEFAULT_IPG;
 		lp->ipg_data.ipg_state = CSTATE;
 	};
